### PR TITLE
Suggest connector templates available at runtime

### LINKED ISSUE #23: Suggest templates available at runtime
### Is your feature request related to a problem? Please describe.

We currently maintain a [list of manually created element template descriptions](https://github.com/bpmn-io/refactorings/blob/main/lib/refactorings/providers/open-ai-element-templates/elementTemplateToolDescriptions.json) that decides whether we can suggest a given element template or not. Since this list is static updating it requires us to release a new version of the `@bpmn-io/refactorings` package.

The implications are:
* only templates for which there is a description in the list can be suggested
* new OOTB templates available at runtime cannot be suggested unless we add a description to the list and release a new version of the package
* user templates cannot be suggested since they can only be known in the context of the Web Modeler
* theoretically, if an existing template is repurposed and has a new description, we'd still suggest it based on the old description

The reason for creating a static list of manually created element template descriptions was that many of the OOTB templates don't have good descriptions resulting in worse performance compared to using the existing template descriptions. But this creates a dependency on [camunda/connectors](https://github.com/camunda/connectors) and means we continually have to come back to this package and update it which is an unreasonable amount of maintenance for such a small feature.

### Describe the solution you'd like

We should remove the static list of element template descriptions and use the descriptions of the element templates available at runtime. Therefore, we should encourage all template authors (Camunda connectors team and Camunda users) to add useful descriptions to their element templates. The problem the LLM faces with low-quality descriptions is essentially the same users face when trying to decide which template to use.

We could also introduce new properties (e.g. `keywords` similar to [`keywords` in package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#keywords)) to aid both conventional search (in [element-template-chooser](https://github.com/bpmn-io/element-template-chooser)) and this feature.

### Describe alternatives you've considered

We could also choose a hybrid approach that uses [curated element template descriptions](https://github.com/bpmn-io/refactorings/blob/main/lib/refactorings/providers/open-ai-element-templates/elementTemplateToolDescriptions.json) and falls back to using an element template's description.

### Additional context

Also found in the context of [assessment - connectors <> core modeling integration](https://miro.com/app/board/uXjVNxSL_wc=/)

### PR DESCRIPTION
**Pull request by philippfromme**
### Proposed Changes

* changes implementation to not rely on static map of descriptions but instead create descriptions from element template descriptions and (optional) keywords
* as a follow-up descriptions of existing connector templates should be overhauled and tags should be added to improve suggestions and add keywords
  * performance will get slightly worse until then

Example of a connector template with description and additional keywords:

```json
{
  "$schema": "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
  "name": "Twitter Connector",
  "id": "io.camunda.connectors.Twitter",
  "version": 1,
  "description": "Post and manage tweets on Twitter",
  "metadata": {
    "keywords": [
      "create tweet",
      "delete tweet",
      "quote tweet",
      "reply to tweet",
      "retweet"
    ]
  },
  "category": {
    "id": "connectors",
    "name": "Connectors"
  },
  "appliesTo": [
    "bpmn:Task"
  ],
  "properties": []
}
```

Closes #23 

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [x] **Visual demo** attached
* [x] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

**Comment by nikku** [comment]
Could you elaborate why you chose "metadata > tags" over "keywords"? Just curious.

**Review by nikku — COMMENTED** [review]
Looks reasonable to me. I was not able to test the change locally.

File: lib/refactorings/providers/open-ai-element-templates/OpenAIElementTemplatesProvider.js [hunk 1]
```diff
@@ -220,4 +208,20 @@ function removeDuplicateRefactorings(refactorings) {
...
+export function idToToolName(id) {
+  return id.replace(/\./g, '_');
+}
+
+export function toolNameToId(name) {
+  return name.replace(/_/g, '.');
+}
+
+function getFunctionDescription(description, tags = []) {
+  if (tags.length) {
```
Comment (nikku) [inline lib/refactorings/providers/open-ai-element-templates/OpenAIElementTemplatesProvider.js]: As this is being read by the LLM I wonder if there is value in additional formatting.

Could be:

```
Description: "SOME DESCRIPTION"

Tags:

- A
- B
- C
- D

Comment (nikku) [reply lib/refactorings/providers/open-ai-element-templates/OpenAIElementTemplatesProvider.js]: I also wonder if we already `""` the description. Seems to not be the case, but _could_ be beneficial.

File: lib/refactorings/providers/open-ai/OpenAIClient.js [hunk 2]
```diff
@@ -78,6 +78,8 @@ export default class OpenAIClient {
         tools
       });
 
+      console.log('response', response);
```
Comment (nikku) [inline lib/refactorings/providers/open-ai/OpenAIClient.js]: Let's remove this debug statement.

Comment (nikku) [reply lib/refactorings/providers/open-ai/OpenAIClient.js]: Alternative: Properly prefix it.

**Comment by nikku** [comment]
@philippfromme This is ready for your second :eyes:. Please feel free to merge as you see fit.

File: lib/refactorings/providers/open-ai-element-templates/OpenAIElementTemplatesProvider.js (hunk 1 above)
Comment (philippfromme) [reply lib/refactorings/providers/open-ai-element-templates/OpenAIElementTemplatesProvider.js]: I tried that and it certainly didn't make the results worse so [let's roll with it.](https://github.com/bpmn-io/refactorings/pull/27/commits/42d5736c52b367527f96bc4d18e5143c2d7746a5)

File: lib/refactorings/providers/open-ai/OpenAIClient.js (hunk 2 above)
Comment (philippfromme) [reply lib/refactorings/providers/open-ai/OpenAIClient.js]: [Dealt with.](https://github.com/bpmn-io/refactorings/pull/27/commits/6a4a472cfdeffaf57730cb8551f614d500acd1fd)

**Comment by philippfromme** [comment]
> Could you elaborate why you chose "metadata > tags" over "keywords"? Just curious.

I renamed it to `keywords`. Same name package.json files use. No particular reason for calling it `tags`.

**Review by nikku — APPROVED** [review]

**Pull request merged**